linuxPackages_latest.virtualbox is failing to build

## Issue description

I'm not able to build VirtualBox with the latest kernel.

### Steps to reproduce

```console
$ nix-build . -A linuxPackages_latest.virtualbox
these derivations will be built:
  /nix/store/b6ykg34ja2cfznhccpmyz93i7g8nwyvs-virtualbox-modules-5.2.22-4.20.8.drv
building '/nix/store/b6ykg34ja2cfznhccpmyz93i7g8nwyvs-virtualbox-modules-5.2.22-4.20.8.drv' on 'ssh://builder@zeus.home.nasreddine.com'...
unpacking sources
unpacking source archive /nix/store/xm1xjw1157fk09jmyvqndj4h64ssn8yq-virtualbox-5.2.22-modsrc
source root is virtualbox-5.2.22-modsrc
patching sources
applying patch /nix/store/r3liap9h7si2hw8rl5ziqd34076m5hgy-fix_kerndir.patch
patching file vboxdrv/Makefile.include.header
Hunk #1 succeeded at 136 with fuzz 1 (offset 19 lines).
patching file vboxnetadp/Makefile.include.header
Hunk #1 succeeded at 136 with fuzz 1 (offset 19 lines).
patching file vboxnetflt/Makefile.include.header
Hunk #1 succeeded at 136 with fuzz 1 (offset 19 lines).
patching file vboxpci/Makefile.include.header
Hunk #1 succeeded at 136 with fuzz 1 (offset 19 lines).
applying patch /nix/store/kv8alkw5nkj3a8a09xs747r90hlkrb0q-fix_kbuild.patch
patching file vboxdrv/Makefile
Hunk #1 succeeded at 132 (offset 1 line).
Hunk #2 succeeded at 158 (offset 1 line).
patching file vboxnetadp/Makefile
Hunk #1 succeeded at 42 (offset 8 lines).
Hunk #2 succeeded at 67 (offset 8 lines).
patching file vboxnetflt/Makefile
Hunk #1 succeeded at 46 (offset 8 lines).
Hunk #2 succeeded at 71 (offset 8 lines).
patching file vboxpci/Makefile
Hunk #1 succeeded at 46 (offset 8 lines).
Hunk #2 succeeded at 68 (offset 8 lines).
configuring
no configure script, doing nothing
building
build flags: -j8 -l8 SHELL=/nix/store/i82x3x0yiijkgyqkzh8ni87gspas0f48-bash-4.4-p23/bin/bash -C /nix/store/vyqh7y4bpv0vq03nf05gwvw9wjwi8f8l-linux-4.20.8-dev/lib/modules/4.20.8/build INSTALL_MOD_PATH=\$\(out\) M=/build/virtualbox-5.2.22-modsrc modules
make: Entering directory '/nix/store/vyqh7y4bpv0vq03nf05gwvw9wjwi8f8l-linux-4.20.8-dev/lib/modules/4.20.8/build'
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxnetadp/linux/VBoxNetAdp-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxnetflt/linux/VBoxNetFlt-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxpci/linux/VBoxPci-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxnetadp/VBoxNetAdp.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxnetflt/VBoxNetFlt.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxnetflt/SUPR0IdcClient.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxpci/VBoxPci.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/linux/SUPDrv-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/SUPDrv.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxnetflt/SUPR0IdcClientComponent.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxnetflt/linux/SUPR0IdcClient-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/SUPDrvGip.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/SUPDrvSem.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxpci/SUPR0IdcClient.o
/build/virtualbox-5.2.22-modsrc/vboxnetadp/linux/VBoxNetAdp-linux.c:132:6: error: 'const struct ethtool_ops' has no member named 'get_settings'; did you mean 'get_strings'?
     .get_settings       = vboxNetAdpEthGetSettings,
      ^~~~~~~~~~~~
      get_strings
/build/virtualbox-5.2.22-modsrc/vboxnetadp/linux/VBoxNetAdp-linux.c:132:27: error: initialization from incompatible pointer type [-Werror=incompatible-pointer-types]
     .get_settings       = vboxNetAdpEthGetSettings,
                           ^~~~~~~~~~~~~~~~~~~~~~~~
/build/virtualbox-5.2.22-modsrc/vboxnetadp/linux/VBoxNetAdp-linux.c:132:27: note: (near initialization for 'gEthToolOpsVBoxNetAdp.get_regs_len')
cc1: some warnings being treated as errors
make[4]: *** [/nix/store/vyqh7y4bpv0vq03nf05gwvw9wjwi8f8l-linux-4.20.8-dev/lib/modules/4.20.8/source/scripts/Makefile.build:291: /build/virtualbox-5.2.22-modsrc/vboxnetadp/linux/VBoxNetAdp-linux.o] Error 1
make[3]: *** [/nix/store/vyqh7y4bpv0vq03nf05gwvw9wjwi8f8l-linux-4.20.8-dev/lib/modules/4.20.8/source/scripts/Makefile.build:516: /build/virtualbox-5.2.22-modsrc/vboxnetadp] Error 2
make[3]: *** Waiting for unfinished jobs....
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxpci/SUPR0IdcClientComponent.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/SUPDrvTracer.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxpci/linux/SUPR0IdcClient-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/SUPLibAll.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/alloc-r0drv.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/initterm-r0drv.o
  LD [M]  /build/virtualbox-5.2.22-modsrc/vboxpci/vboxpci.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/memobj-r0drv.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/mpnotification-r0drv.o
  LD [M]  /build/virtualbox-5.2.22-modsrc/vboxnetflt/vboxnetflt.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/powernotification-r0drv.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/assert-r0drv-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/alloc-r0drv-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/initterm-r0drv-linux.o
/build/virtualbox-5.2.22-modsrc/vboxdrv/SUPDrvTracer.o: warning: objtool: .text+0x7: indirect jump found in RETPOLINE build
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/memobj-r0drv-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/memuserkernel-r0drv-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/mp-r0drv-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/mpnotification-r0drv-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/process-r0drv-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/rtStrFormatKernelAddress-r0drv-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/semevent-r0drv-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/semeventmulti-r0drv-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/semfastmutex-r0drv-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/semmutex-r0drv-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/spinlock-r0drv-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/thread-r0drv-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/thread2-r0drv-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/threadctxhooks-r0drv-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/time-r0drv-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/timer-r0drv-linux.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/generic/semspinmutex-r0drv-generic.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/common/alloc/alloc.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/common/checksum/crc32.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/common/checksum/ipv4.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/common/checksum/ipv6.o
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/common/err/RTErrConvertFromErrno.o
/build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/time-r0drv-linux.c: In function 'VBoxHost_RTTimeNow':
/build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/time-r0drv-linux.c:175:5: error: implicit declaration of function 'ktime_get_real_ts'; did you mean 'ktime_get_real_ns'? [-Werror=implicit-function-declaration]
     ktime_get_real_ts(&Ts);
     ^~~~~~~~~~~~~~~~~
     ktime_get_real_ns
  CC [M]  /build/virtualbox-5.2.22-modsrc/vboxdrv/common/err/RTErrConvertToErrno.o
cc1: some warnings being treated as errors
make[4]: *** [/nix/store/vyqh7y4bpv0vq03nf05gwvw9wjwi8f8l-linux-4.20.8-dev/lib/modules/4.20.8/source/scripts/Makefile.build:291: /build/virtualbox-5.2.22-modsrc/vboxdrv/r0drv/linux/time-r0drv-linux.o] Error 1
make[4]: *** Waiting for unfinished jobs....
make[3]: *** [/nix/store/vyqh7y4bpv0vq03nf05gwvw9wjwi8f8l-linux-4.20.8-dev/lib/modules/4.20.8/source/scripts/Makefile.build:516: /build/virtualbox-5.2.22-modsrc/vboxdrv] Error 2
make[2]: *** [/nix/store/vyqh7y4bpv0vq03nf05gwvw9wjwi8f8l-linux-4.20.8-dev/lib/modules/4.20.8/source/Makefile:1562: _module_/build/virtualbox-5.2.22-modsrc] Error 2
make[1]: *** [Makefile:152: sub-make] Error 2
make: *** [Makefile:15: __sub-make] Error 2
make: Leaving directory '/nix/store/vyqh7y4bpv0vq03nf05gwvw9wjwi8f8l-linux-4.20.8-dev/lib/modules/4.20.8/build'
builder for '/nix/store/b6ykg34ja2cfznhccpmyz93i7g8nwyvs-virtualbox-modules-5.2.22-4.20.8.drv' failed with exit code 2
error: build of '/nix/store/b6ykg34ja2cfznhccpmyz93i7g8nwyvs-virtualbox-modules-5.2.22-4.20.8.drv' on 'ssh://builder@zeus.home.nasreddine.com' failed: builder for '/nix/store/b6ykg34ja2cfznhccpmyz93i7g8nwyvs-virtualbox-modules-5.2.22-4.20.8.drv' failed with exit code 2
builder for '/nix/store/b6ykg34ja2cfznhccpmyz93i7g8nwyvs-virtualbox-modules-5.2.22-4.20.8.drv' failed with exit code 1
error: build of '/nix/store/b6ykg34ja2cfznhccpmyz93i7g8nwyvs-virtualbox-modules-5.2.22-4.20.8.drv' failed
```

## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 4.20.7, NixOS, 19.03pre-git (Koi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2`
 - channels(root): `"nixos-19.03pre167858.f2a1a4e93be"`
 - nixpkgs: `/run/current-system/nixpkgs`

--- a/pkgs/applications/virtualization/virtualbox/default.nix
+++ b/pkgs/applications/virtualization/virtualbox/default.nix
@@ -19,9 +19,10 @@ with stdenv.lib;
 let
   python = python2;
   buildType = "release";
-  # Remember to change the extpackRev and version in extpack.nix as well.
-  main = "1m48ywa913g6zgqslvrihxs2fbr4gmljypbdpjma2hck6isyi02m";
-  version = "5.2.22";
+  # Remember to change the extpackRev and version in extpack.nix and
+  # guest-additions/default.nix as well.
+  main = "0rylf1g0vmv0q19iyvyq4dj5h9yvyqqnmmqaqrx93qrv8s1ybssd";
+  version = "5.2.26";
 in stdenv.mkDerivation {
   name = "virtualbox-${version}";
 

--- a/pkgs/applications/virtualization/virtualbox/extpack.nix
+++ b/pkgs/applications/virtualization/virtualbox/extpack.nix
@@ -2,7 +2,7 @@
 
 with lib;
 
-let version = "5.2.22";
+let version = "5.2.26";
 in
 fetchurl rec {
   name = "Oracle_VM_VirtualBox_Extension_Pack-${version}.vbox-extpack";
@@ -10,7 +10,7 @@ fetchurl rec {
   sha256 =
     # Manually sha256sum the extensionPack file, must be hex!
     # Thus do not use `nix-prefetch-url` but instead plain old `sha256sum`.
-    let value = "779250666551b2f5426e86c2d21ceb0209b46174536971611025f753535131ef";
+    let value = "4b7caa9b722840d49f154c3e5efb6463b1b7129f09973a25813dfdbccd9debb7";
     in assert (builtins.stringLength value) == 64; value;
 
   meta = {

--- a/pkgs/applications/virtualization/virtualbox/guest-additions/default.nix
+++ b/pkgs/applications/virtualization/virtualbox/guest-additions/default.nix
@@ -19,7 +19,7 @@ stdenv.mkDerivation {
 
   src = fetchurl {
     url = "http://download.virtualbox.org/virtualbox/${version}/VBoxGuestAdditions_${version}.iso";
-    sha256 = "e51e33500a265b5c2d7bb2d03d32208df880523dfcb1e2dde2c78a0e0daa0603";
+    sha256 = "0f3w9wjd5aj2khzqh37vdg86wqbbx4gx9aidaai9syn9sk8ca9xr";
   };
 
   KERN_DIR = "${kernel.dev}/lib/modules/${kernel.modDirVersion}/build";

